sound/soc/intel/boards/broadwell.c

```diff
@@ -218,7 +225,11 @@ static struct snd_soc_dai_link broadwell_rt286_dais[] = {
 		.ops = &broadwell_rt286_ops,
 		.dpcm_playback = 1,
 		.dpcm_capture = 1,
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_BROADWELL)
+		SND_SOC_DAILINK_REG(ssp0_pin, codec, platform),
+#else
 		SND_SOC_DAILINK_REG(dummy, codec, dummy),
+#endif
```

Comment:
why is this needed?
we've managed to make things work in the past without this change.
What is the problem and what are you fixing here?

Reply:
@plbossart The old one what to map to a dummy BE, but we only have a dummy BE with nocodec support.
Now nocodec is default disabled with HDA link enabled, so the change is required.

Or we should add dummy BE to SOF BDW platform drvier.

Reply:
@xiulipan with the nocodec machine drv, we dont use the dummy CPU DAI, we actually use the ones defined in the dai_drv[]. We use dummy dai only for the codec dai. 

Anyway, how does it work with the broadwell legacy driver with the dummy CPU DAI? We never register the dummy codec DAI in SOF. So why isnt the dummy CPU DAI already registered too?

Reply:
Not sure, but from the test and log, it says
```
[  217.561885] broadwell-audio broadwell-audio: info: override BE DAI link Codec
[  217.561898] broadwell-audio broadwell-audio: ASoC: CPU DAI snd-soc-dummy-dai not registered
[  218.048635] sof-audio-acpi INT3438:00: ipc rx: 0x90020000: GLB_TRACE_MSG
```

And we do have something in nocodec
https://github.com/thesofproject/linux/blob/fd30230359e5b9106d69eb3cf30ef987921d2cee/sound/soc/sof/nocodec.c#L53-L54